pkg/sql/pg_catalog.go

```diff
@@ -1436,8 +1436,43 @@ CREATE TABLE pg_catalog.pg_enum (
   enumsortorder FLOAT4,
   enumlabel STRING
 )`,
-	populate: func(_ context.Context, p *planner, _ *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
-		// Enum types are not currently supported.
+	populate: func(ctx context.Context, p *planner, dbContext *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		h := makeOidHasher()
+		descs, err := p.Tables().getAllDescriptors(ctx, p.txn)
+		if err != nil {
+			return err
+		}
+		for _, desc := range descs {
+			typDesc, ok := desc.(*sqlbase.TypeDescriptor)
+			if !ok {
+				continue
+			}
+			if dbContext != nil && typDesc.ParentID != dbContext.ID {
+				continue
+			}
+			if typDesc.Kind != sqlbase.TypeDescriptor_ENUM {
+				continue
+			}
+			// Generate a row for each member of the enum. We don't represent enums
+			// internally using floats for ordering like Postgres, so just pick a
+			// float entry for the rows.
+			typ := types.MakeEnum(uint32(typDesc.ID))
+			if err := typDesc.HydrateTypeInfo(typ); err != nil {
+				return err
+			}
+			enumData := typ.TypeMeta.EnumData
+			typOID := tree.NewDOid(tree.DInt(typ.Oid()))
+			for i := range enumData.LogicalRepresentations {
+				if err := addRow(
+					h.EnumEntryOid(typOID, enumData.PhysicalRepresentations[i]),
```

Comment:
does this change every time the ENUM changes? if so that seems bad.
can we use some sort of other unique identifier? `typDesc.ID` seems better.

Reply:
No, the physical representations are static. Either way, IDT we could do typDesc.ID because each row here needs a new OID.

Reply:
oh right, i totally missed that this is happening on each logical repr.